webgui-new/next.config.js

```diff
@@ -0,0 +1,18 @@
+require('dotenv').config();
+
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  reactStrictMode: true,
+  modularizeImports: {
+    '@mui/icons-material': {
+      transform: '@mui/icons-material/{{member}}',
+    },
+  },
+  trailingSlash: true,
+  ...(process.env.NODE_ENV === 'production' && {
+    assetPrefix: '/new',
+    basePath: '/new',
```

Comment:
It is not guaranteed that the new web GUI will be served on the `/new` absolute path, e.g. it can be served on `/demo/new` here:
https://codecompass.net/demo/new/

This required the local modification of this file. I am thinking about whether NextJs could handle relative paths here or not?

Reply:
This is only a prefix for Next.js requests and it does not effect the webserver path. It is required because the exported contents will be in the `/new` folder of the webserver (in this case a symbolic link). So Next.js needs to know to look for the files in this folder, and not at the root of the webserver.

Reply:
No, it does effect the webserver path. Eg. on the *codecompass.net* server we are serving the new frontend on the https://codecompass.net/demo/new/ path. If these variables have the value `/new` before `npm run build`, then the frontend will not work, because it assumes that the base path is `https://codecompass.net/new/`, which is incorrect. Therefore we had to modify the values of `assetPrefix` and `basePath` to `/demo/new` and rebuild the frontend.

Therefore the values here should not be hardcoded. If a relative path could be supported, that would be the best. Otherwise it should be configurable through an environment variable (usually named `PUBLIC_URL` in React apps).

Reply:
I see. I could not test it with a webserver that contains additional paths, so I did not know about this. In that case, I will add another `.env` variable for this, I don't think relative paths are supported here.